### PR TITLE
Support takeElementScreenshot for o11y

### PR DESCRIPTION
**Pull request by samarsault**

### CODE REVIEW

File: src/testObservability.js [hunk 1]
```diff
@@ -229,6 +229,8 @@ class TestObservability {
       for (const command of eventData.commands) {
         if (command.name === 'saveScreenshot' && command.args) {
           await this.createScreenshotLogEvent(testUuid, command.args[0], command.startTime);
+        } else if (/^.*(takeElementScreenshot)$/.test(command.name) && command.result && command.result.valuePath) {
```
Comment (sauravdas1997) [inline src/testObservability.js:232]: `.contains` instead if regex ?

Comment (sauravdas1997) [inline src/testObservability.js:232]: If the bytesize of <500 then it will be value and not valuePath

Comment (sauravdas1997) [inline src/testObservability.js:232]: optional chaining ? `command?.result?.valuePath`

**Comment by sauravdas1997** [comment]
This feature is not enabled by default, we have to add the flag in nightwatch_conf as well

**Review by sauravdas1997 — CHANGES_REQUESTED** [review]

File: src/testObservability.js (hunk 1 above)
Comment (samarsault) [reply src/testObservability.js:232]: Won't work in node 12 and NW supports 12.0+

Comment (samarsault) [reply src/testObservability.js:232]: regex is more accurate here and performance difference is minimal

Comment (samarsault) [reply src/testObservability.js]: addressed

File: src/testObservability.js [hunk 2]
```diff
@@ -228,7 +228,20 @@ class TestObservability {
...
+            const screenshot = fs.readFileSync(screenshotPath, 'base64');
+            await this.createScreenshotLogEvent(testUuid, screenshot, command.startTime);
+          }
+        } else if (/^.*(takeElementScreenshot)$/.test(command.name) && command.result) {
+          if (command.result.value) {
+            await this.createScreenshotLogEvent(testUuid, command.result.value, command.startTime);
+          } else if (command.result.valuePath) {
+            if (fs.existsSync(command.result.valuePath)) {
+              const screenshot = fs.readFileSync(command.result.valuePath, 'utf8');
+              await this.createScreenshotLogEvent(testUuid, screenshot, command.startTime);
```
Comment (07souravkunda) [inline src/testObservability.js:242]: add graceful error handling

File: src/testObservability.js [hunk 3]
```diff
@@ -228,7 +228,20 @@ class TestObservability {
     if (process.env.BS_TESTOPS_ALLOW_SCREENSHOTS === 'true') {
       for (const command of eventData.commands) {
         if (command.name === 'saveScreenshot' && command.args) {
-          await this.createScreenshotLogEvent(testUuid, command.args[0], command.startTime);
+          const screenshotPath = command.args[0];
+          if (fs.existsSync(screenshotPath)) {
+            const screenshot = fs.readFileSync(screenshotPath, 'base64');
+            await this.createScreenshotLogEvent(testUuid, screenshot, command.startTime);
+          }
+        } else if (/^.*(takeElementScreenshot)$/.test(command.name) && command.result) {
```
Comment (07souravkunda) [inline src/testObservability.js:236]: add this regex to constants.
Add a comment why doing this

**Review by 07souravkunda — CHANGES_REQUESTED** [review]

File: src/testObservability.js [hunk 4]
```diff
@@ -228,7 +228,20 @@ class TestObservability {
...
+          if (command.result.value) {
```
Comment (07souravkunda) [inline src/testObservability.js:237]: Use nestedKey if you are unsure about whether the `result` exists.

**Review by 07souravkunda — CHANGES_REQUESTED** [review]

File: src/testObservability.js (hunk 4 above)
Comment (sauravdas1997) [inline src/testObservability.js:237]: use optional params

**Comment by sauravdas1997** [comment]
Nightwatch themselves uses optional chaining, so IMO should be okay to use that

**Comment by samarsault** [comment]
We dont need optional chaining since we have to check the if path

File: src/testObservability.js (hunk 4 above)
Comment (samarsault) [reply src/testObservability.js:237]: We have a check earlier

File: src/testObservability.js (hunk 3 above)
Comment (samarsault) [reply src/testObservability.js]: done

File: src/testObservability.js (hunk 2 above)
Comment (samarsault) [reply src/testObservability.js:235]: done

**Review by 07souravkunda — APPROVED** [review]

**Review by sauravdas1997 — APPROVED** [review]

**Review by sauravdas1997 — APPROVED** [review]

**Pull request merged**